cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -705,14 +706,19 @@ withIndexEntries verbosity (RepoIndex _repoCtxt (RepoLocalNoIndex (LocalRepo nam
 
     entries <- handle handler $ fmap catMaybes $ for dirContents $ \file -> do
         case isTarGz file of
-            Nothing -> do
-                unless (takeFileName file == "noindex.cache" || ".cabal" `isSuffixOf` file) $
-                    info verbosity $ "Skipping " ++ file
-                return Nothing
+            Nothing
+              | takeFileName file == "preferred-versions" -> do
+                  contents <- BS.readFile (localDir </> file)
+                  let versionPreferences = parsePreferredVersions contents
```

Comment:
I'd leave atl least a note that `parsePreferredVersions` never fails. That **will** confuse users, as there is no error reporting. (It's somewhat fine that we trust that Hackage generated index is ok, users: we couldn't).

I'd rather write something which at least warns.

Reply:
That is a very good point, will update this logic

Reply:
Included warnings